OpenRA.Mods.Common/Activities/Resupply.cs

```diff
@@ -17,48 +17,94 @@
 
 namespace OpenRA.Mods.Common.Activities
 {
-	public class Repair : Activity
+	public class Resupply : Activity
 	{
 		readonly IHealth health;
 		readonly RepairsUnits[] allRepairsUnits;
 		readonly Target host;
 		readonly WDist closeEnough;
 		readonly Repairable repairable;
+		readonly Rearmable rearmable;
+		readonly INotifyResupply[] notifyResupplies;
 
 		int remainingTicks;
-		bool played = false;
+		bool played;
+		bool paused;
+		bool repairComplete;
+		bool rearmComplete;
+		ResupplyType activeResupplyTypes = ResupplyType.None;
 
-		public Repair(Actor self, Actor host, WDist closeEnough)
+		public Resupply(Actor self, Actor host, WDist closeEnough)
 		{
 			this.host = Target.FromActor(host);
 			this.closeEnough = closeEnough;
 			allRepairsUnits = host.TraitsImplementing<RepairsUnits>().ToArray();
 			health = self.TraitOrDefault<IHealth>();
 			repairable = self.TraitOrDefault<Repairable>();
+			rearmable = self.TraitOrDefault<Rearmable>();
+			notifyResupplies = host.TraitsImplementing<INotifyResupply>().ToArray();
+			repairComplete = health == null || health.DamageState == DamageState.Undamaged
+				|| repairable == null || !repairable.Info.RepairActors.Contains(host.Info.Name) || !allRepairsUnits.Any();
```

Comment:
This breaks for actors that only have `RepairableNear`.

Reply:
Oops, overlooked that II.

Reply:
Fixed, but considering what changes I had to make to fix this, I'm a little baffled `RepairableNear` even works on bleed at all, as the Repair activity also only checked `Repairable`.
Well, not really worth investigating further, `RepairableNear` will go away soon, anyway.